### DISCUSSION TITLE
Microsoft OAuth Fetching Mails are very slow

### DISCUSSION THREAD

**Discussion by HochzeitManagement** [jstedfast/MailKit#1732] [General]
We are experiencing significant issues with mailboxes that come from Microsoft Exchange OAuth. If I want to synchronize a folder with 10,000 emails, it takes forever. 

How should we proceed?
We fetch 300 emails, and this usually takes about 30 seconds. Just for fetching. 
![image](https://github.com/jstedfast/MailKit/assets/61002651/4c533fde-76d5-4122-9064-1aef561333fe)

Why is it so slow? 
Are we doing something wrong here? Or is the connection just that slow?
With other providers, this is super fast.

Thanks for your help

**Comment by jstedfast** [comment]
If you enable protocol-logging by passing a ProtocolLogger with timestamps enabled to the ImapClient .ctor, does it show the response from the server taking a long time to send and/or each individual response taking a while? Or?

As you noted, MailKit is very fast with other servers, so it sounds like this is just a slow connection to Microsoft Exchange -or- it's just the server being slow to process requests.
